engines/buried/saveload.cpp

```diff
@@ -47,12 +47,26 @@ namespace Buried {
 
 #define SAVEGAME_CURRENT_VERSION 1
 
-bool BuriedEngine::canLoadGameStateCurrently() {
-	return !isDemo() && _mainWindow && !_yielding;
+bool BuriedEngine::canLoadGameStateCurrently(Common::U32String *msg) {
+	if (isDemo()) {
+		if (msg)
+			*msg = _("This game does not support loading");
+
+		return false;
+	}
+
+	return _mainWindow && !_yielding;
 }
 
-bool BuriedEngine::canSaveGameStateCurrently() {
-	return !isDemo() && _mainWindow && !_yielding && ((FrameWindow *)_mainWindow)->isGameInProgress();
+bool BuriedEngine::canSaveGameStateCurrently(Common::U32String *msg) {
+	if (isDemo()) {
+		if (msg)
+			*msg = _("This game does not support saving");
```

Comment:
It would be preferable to have these default messages placed somewhere globally, where they can be reused and avoid potential typos. Perhaps we could place them as static const strings inside the Engine class?

Reply:
These are not default. The default ones sit in `engine/dialogs.cpp`

Reply:
What I meant is that these messages are generic enough ("This game does not support saving"), so they could be reused among engines (which would also help avoid typos)

Reply:
Do you mean something like this?

to the `engines/engine.h`:

```C++
	const char *kCanLoadReasonUnsupported = _s("This game does not support loading");
```

`asylym/asylum.cpp`:

```C++
	if (checkGameVersion("Demo")) {
		if (msg)
			*msg = _(Common::kCanLoadReasonUnsupported);

		return false;
	}
```

Isn't that ugly?